cmd/space-ipfs-node/main.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"sync"
+
+	icore "github.com/ipfs/interface-go-ipfs-core"
+	peerstore "github.com/libp2p/go-libp2p-peerstore"
+	ma "github.com/multiformats/go-multiaddr"
+
+	"github.com/ipfs/go-ipfs-config"
+	"github.com/ipfs/go-ipfs/core/node/libp2p"
+	"github.com/ipfs/go-ipfs/commands"
+	"github.com/ipfs/go-ipfs/core"
+	"github.com/ipfs/go-ipfs/core/coreapi"
+	"github.com/ipfs/go-ipfs/core/corehttp"
+	"github.com/ipfs/go-ipfs/plugin/loader"
+	"github.com/ipfs/go-ipfs/repo/fsrepo"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+
+func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	repoPath, err := config.PathRoot()
+	if err != nil {
+		panic(err)
+	}
+
+	if err := setupPlugins(repoPath); err != nil {
+		panic(err)
+	}
+
+	// open the repo
+	repo, err := fsrepo.Open(repoPath)
+	if err != nil {
+		panic(err)
+	}
+
+	// construct the node
+	nodeOptions := &core.BuildCfg{
+		Online:  true,
+		Routing: libp2p.DHTClientOption,
+		Repo: repo,
+	}
+
+	node, err := core.NewNode(ctx, nodeOptions)
+	if err != nil {
+		panic(err)
+	}
+
+	core, err := coreapi.NewCoreAPI(node)
```

Comment:
If there is no existing node, we should call the logic behind `ipfs init` to setup the node (ran this and it complained about no ipfs init).